packages/angular_devkit/build_angular/src/builders/karma/tests/setup.ts

```diff
@@ -27,3 +42,133 @@ export const BASE_OPTIONS = Object.freeze<Schema>({
   progress: false,
   watch: false,
 });
+
+const optionSchemaCache = new Map<string, json.schema.JsonSchema>();
```

Comment:
This code is _similar_ to the dev-server code and could likely be refactored to be shared across builders: https://github.com/angular/angular-cli/blob/main/packages/angular_devkit/build_angular/src/builders/dev-server/tests/jasmine-helpers.ts#L18-L25